### PR TITLE
Update scripts to use -m mypy

### LINKED ISSUE #548: Cygwin awkwardness
Current cygwin python3 is 3.2.5, so I install and use the native Windows Python which is 3.4.2 installed at `C:\Python34` (the default).

After installing mypy via `python setup.py instal`" it installs under `C:\Python34`, as expected, and creates a `C:\Python34\Scripts\mypy` file.  Note there is no `mypy.exe` program.  (There are such programs for `pip.exe` and `easy_install.exe`, would be helpful for mypy to do likewise.)

To run this under cygwin requires an explicit call to `python` (bash turns the script path into a unix-y one, which confuses native Windows python -- not your issue, I mention this for clarity), this:

``` bash
$ export PATH=/cygdrive/c/Python34:$PATH
$ python C:/Python34/Scripts/mypy my-file.py
```

Oddly this complains there is no `enum` module to import, though running `python my-file.py` directly works fine.  Running the same under a native CMD prompt behaves the same.

I'm unsure what about running from a Cygwin bash shell is causing the issue, perhaps it isn't mypy's fault at all but I'm at a loss how to investigate.

### PR DESCRIPTION
**Pull request by Igonato**
This fixes #548

Tested with CPython on:

 - Windows
   - cygwin shell
   - cygwin with virtualenv
   - cmd
   - cmd with virtualenv
- Ubuntu 14.04
   - bash
   - bash virtualenv

### CODE REVIEW

**Comment by rtxm** [comment]
The title is quite misleading: you wrote 'numpy' instead of 'mypy' :wink:

**Comment by Igonato** [comment]
@rtxm don't get enough sleep lately xD. I should I rebase to fix the commit message as well, I guess

File: scripts/mypy [hunk 1]
```diff
@@ -1,6 +1,8 @@
-#!/usr/bin/env python3
-"""Mypy type checker command line tool."""
+#!/usr/bin/env bash
```
Comment (gvanrossum) [inline scripts/mypy]: I don't feel strongly about what you're doing with Windows, but I feel the changes to this file are not an improvement. How do you even know the relationship between the python binary and this script?

**Comment by Igonato** [comment]
@gvanrossum nevermind, I think I jumped the gun on it. On top of these, it requires a number of changes in the testing suite, which isn't an improvement either. I guess It's fine as is, it works as `mypy.bat stuff`, it works as `python -m mypy stuff`, that's should be enough for Cygwin users.

**Pull request closed without merging**